### ISSUE TITLE
ServiceInspectWithRaw() doesn't return a version number, which makes replica scaling impossible with ServiceUpdate()

### ISSUE THREAD

**Issue by harrisonturton** [moby/moby#38047]
Hello,

I was running into an [issue](https://github.com/moby/moby/issues/30794#issuecomment-429741548) when trying to update a service through the Go SDK. I need to pass in a version number.

This number isn't present on `docker service inspect --pretty`, but it exists on `docker service inspect`. This isn't very helpful however, since it is not included in [swarm.Service](https://godoc.org/github.com/docker/docker/api/types/swarm#Service), which is returned by `client.ServiceInspectWithRaw()`.

Basically, this stops me from scaling the number of replicas with `client.ServiceUpdate()`, since I need to pass in a version number.

It would be helpful if this version number is included in [swarm.Service](https://godoc.org/github.com/docker/docker/api/types/swarm#Service), or alternatively somewhere in [swarm.ServiceSpec](https://godoc.org/github.com/docker/docker/api/types/swarm#ServiceSpec).

In the meantime, is there a way around this? How can I find that version number?

Cheers.

**Comment by harrisonturton** [comment]
When attempting a pull request, I realise I made a mistake! `client.ServiceInspectWithRaw()` returns the following `swarm.Service`:

```
type Service struct {
	ID string
	Meta
	Spec         ServiceSpec   `json:",omitempty"`
	PreviousSpec *ServiceSpec  `json:",omitempty"`
	Endpoint     Endpoint      `json:",omitempty"`
	UpdateStatus *UpdateStatus `json:",omitempty"`
}
```

I had brushed over the `Meta` field, which has a `Version` field:

```
type Meta struct {
    Version   Version   `json:",omitempty"`
    CreatedAt time.Time `json:",omitempty"`
    UpdatedAt time.Time `json:",omitempty"`
}
```

Sorry about this – I should have dug through the docs more. Maybe we should make this more obvious in the documentation for `swarm.ServiceUpdate()`?

**Comment by harrisonturton** [comment]
When attempting a pull request, I realise I made a mistake! `client.ServiceInspectWithRaw()` returns the following `swarm.Service`:

```
type Service struct {
	ID string
	Meta
	Spec         ServiceSpec   `json:",omitempty"`
	PreviousSpec *ServiceSpec  `json:",omitempty"`
	Endpoint     Endpoint      `json:",omitempty"`
	UpdateStatus *UpdateStatus `json:",omitempty"`
}
```

I had brushed over the `Meta` field, which has a `Version` field:

```
type Meta struct {
    Version   Version   `json:",omitempty"`
    CreatedAt time.Time `json:",omitempty"`
    UpdatedAt time.Time `json:",omitempty"`
}
```

Sorry about this – I should have dug through the docs more. Maybe we should make this more obvious in the documentation for `swarm.ServiceUpdate()`?

**Comment by thaJeztah** [comment]
Glad you found it! If you have some ideas to improve the docs for this; pull requests are definitely welcome ☺️🤗

**Comment by harrisonturton** [comment]
@thaJeztah Happy to do so! Where is the repo for the [Go SDK](https://godoc.org/github.com/docker/docker/client#) documentation? I can't seem to find it.

**Comment by thaJeztah** [comment]
@harrisonturton documentation at https://godoc.org/github.com/docker/docker/client  is generated from the Go source code in this repository. For example, the package description shown there is coming from this comment (GoDoc); https://github.com/moby/moby/blob/de53bbf92b7968f6b647dd7cf074095acc1d7388/client/client.go#L2-L39

So, to update that documentation, the comments in the source code have to be updated, which will automatically be reflected on godoc.org. You can find some information about GoDoc on the Golang blog; https://blog.golang.org/godoc-documenting-go-code
